src/cmd/ksh93/sh/nvtype.c

```diff
@@ -1568,8 +1568,8 @@ Namval_t *nv_typeparent(Namval_t *np) {
 }
 
 void nv_checkrequired(Namval_t *mp) {
-    Namtype_t *dp, *dq = 0;
-    Namval_t *np, *mq, *nq;
+    Namtype_t *dp = NULL, *dq = NULL;
```

Comment:
It's a shame C doesn't have something like C++11's `nullptr`. So while changing `0` to `NULL` has no practical effect we should be making such changes anywhere the value is meant to be interpreted as a pointer if only for clarity to a human reader.

Reply:
LGTM